ambry-router/src/test/java/com.github.ambry.router/GetBlobOperationTest.java

```diff
@@ -467,17 +467,40 @@ public void testSuccessInThePresenceOfVariousErrors() throws Exception {
     Properties props = getDefaultNonBlockingRouterProperties();
     props.setProperty("router.datacenter.name", "DC1");
     routerConfig = new RouterConfig(new VerifiableProperties(props));
-    testVariousErrors(dcWherePutHappened);
+    doTestSuccessInThePresenceOfVariousErrors(dcWherePutHappened);
 
     props = getDefaultNonBlockingRouterProperties();
     props.setProperty("router.datacenter.name", "DC2");
     routerConfig = new RouterConfig(new VerifiableProperties(props));
-    testVariousErrors(dcWherePutHappened);
+    doTestSuccessInThePresenceOfVariousErrors(dcWherePutHappened);
 
     props = getDefaultNonBlockingRouterProperties();
     props.setProperty("router.datacenter.name", "DC3");
     routerConfig = new RouterConfig(new VerifiableProperties(props));
-    testVariousErrors(dcWherePutHappened);
+    doTestSuccessInThePresenceOfVariousErrors(dcWherePutHappened);
+  }
+
+  /**
+   * Tests the case where all servers return the same server level error code
+   * @throws Exception
+   */
+  @Test
+  public void testFailureOnServerErrors() throws Exception {
+    doPut();
+    List<ServerErrorCode> serverErrors = new ArrayList<>(Arrays.asList(ServerErrorCode.values()));
```

Comment:
Can use `EnumSet` here too, unless the order is important.

Reply:
It is not - done